expression/constant_propagation.go

```diff
@@ -50,21 +50,125 @@ func (m *multiEqualSet) findRoot(a int) int {
 	return m.parent[a]
 }
 
-type propagateConstantSolver struct {
-	colMapper  map[string]int // colMapper maps column to its index
-	unionSet   *multiEqualSet // unionSet stores the relations like col_i = col_j
-	eqList     []*Constant    // if eqList[i] != nil, it means col_i = eqList[i]
-	columns    []*Column      // columns stores all columns appearing in the conditions
+type basePropConstSolver struct {
+	colMapper map[string]int // colMapper maps column to its index
+	eqList    []*Constant    // if eqList[i] != nil, it means col_i = eqList[i]
+	unionSet  *multiEqualSet // unionSet stores the relations like col_i = col_j
+	columns   []*Column      // columns stores all columns appearing in the conditions
+	ctx       sessionctx.Context
+}
+
+func (s *basePropConstSolver) getColID(col *Column) int {
+	code := col.HashCode(nil)
```

Comment:
HashCode is actually hash the `UniqueID` to byte slice. You can use it directly.

Reply:
Make sense, will do it.